tools/nl/embeddings/integrity_test.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+
+from nl_server import config_reader
+from shared.lib import gcs
+from tools.nl.embeddings import utils
+
+
+class TestIntegrity(unittest.TestCase):
+  """Check that the index in catalog.yaml is up to date.
+
+  Read the md5sum of each index index and check it against the _preindex.csv in
+  the input directory.
+  """
+
+  def test_md5sum(self):
+    catalog = config_reader.read_catalog
+
+    for embeddings_name, index_config in catalog().indexes.items():
+      if embeddings_name == 'medium_ft':
+        continue
+      if index_config.store_type == 'MEMORY':
+        embeddings_dir = os.path.dirname(index_config.embeddings_path)
+      elif index_config.store_type == 'LANCEDB':
+        embeddings_dir = index_config.embeddings_path
+      else:
+        raise ValueError('Unsupported store type: %s' % index_config.store_type)
+      md5sum_path = os.path.join(embeddings_dir, 'md5sum.txt')
+      if gcs.is_gcs_path(md5sum_path):
+        md5sum_path = gcs.maybe_download(md5sum_path)
+      with open(md5sum_path) as f:
+        got_md5sum = f.read().strip()
+      expected_md5sum = utils.get_md5sum(
+          os.path.join(index_config.source_path, '_preindex.csv'))
+      self.assertEqual(got_md5sum, expected_md5sum)
```

Comment:
What use-case do you anticipate for md5 hash?  Sanity-check against someone checking in a wrong path?  Or actual content shift of some form?

If the former, this LG.  But if latter, I slightly wonder if we should just put the whole _prefix.csv and compare, so the diffs can make sense?

Reply:
How I start this is finding that different indexes using the same descriptions might not be updated at the same time.
For example, after changing base descriptions, the lancedb and mistral index are not updated. 
This was a problem (but not anymore) when undata and ilo also relies on base/sdg description.

Right now, can make sure the _preindex of a folder is indeed what is in the actual embeddings.csv for all the indexes. And this is necessary for increment update..

Reply:
checking in the _preindex.csv also works but i am a bit worried that the gcs download of many ( potentially large) files can make the test heavy.